pyproject.toml

```diff
@@ -46,7 +46,7 @@ Pillow = ">=10.0.1,<11"
 retrying = ">=1.3.3,<2"
 Shapely = ">=1.8.5,<3"
 termcolor = ">=1.1,<3"
-pyarrow = ">=8"
+pyarrow = ">=8,<17"
```

Comment:
nit: a comment for awareness and tracking?

Reply:
this does not fix the issue, still investigating what's going on. I suspect the error could be caused by pyarrow v17 being released today.

```
 RuntimeError

  Unable to find installation candidates for pyarrow (17.0.0)

  at /opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/poetry/installation/chooser.py:76 in choose_for
       72│ 
       73│             links.append(link)
       74│ 
       75│         if not links:
    →  76│             raise RuntimeError(f"Unable to find installation candidates for {package}")
       77│ 
       78│         # Get the best link
       79│         chosen = max(links, key=lambda link: self._sort_key(package, link))
       ```